.github/workflows/binaries.yml

```diff
@@ -0,0 +1,61 @@
+name: Generate binaries
+on:
+  release:
+    types:
+      - created
+jobs:
+  binaries:
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            asset_name: api-linux-amd64.tar.gz
+            compress_cmd: tar -czvf
+            build_cmd: docker run -v $PWD:/data golang:1.19 bash -c "cd /data && go build ./cmd/api"
+          - os: windows-latest
+            asset_name: api-windows-amd64.zip
+            compress_cmd: tar.exe -a -c -f
+            build_cmd: go build -o api ./cmd/api
+          - os: macos-latest
+            asset_name: api-darwin-amd64.tar.gz
+            compress_cmd: tar -czvf
+            build_cmd: go build -o api ./cmd/api
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: v1.19.x
+      - name: Build binary
+        run: ${{ matrix.build_cmd }}
+      - name: Pack output
+        run: ${{ matrix.compress_cmd }} ${{ matrix.asset_name }} api
+      - name: Upload binary
+        uses: svenstaro/upload-release-action@v2
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: ${{ matrix.asset_name }}
+          asset_name: ${{ matrix.asset_name }}
+          tag: ${{ github.ref_name }}
+          overwrite: true
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
```

Comment:
Just in case: QEMU is a machine emulator/virtualizer. So you can run binaries from other architectures in your architecture. Of course, the run will be way slower than running natively, but it's useful to do multi-platform image creations.